python/lambdas/pause-preservica-activity/test_pause_preservica_activity.py

```diff
@@ -0,0 +1,187 @@
+    monkeypatch.setenv('ENVIRONMENT', 'test')
+
+
+@pytest.fixture
+def boto3_mocks():
+    with patch('pause_preservica_activity.eventbridge') as eventbridge_mock, \
+            patch('pause_preservica_activity.secretsmanager') as secretsmanager_mock:
+        yield eventbridge_mock, secretsmanager_mock
+
+
+def test_pause_preservica_activity(env, boto3_mocks):
+    eventbridge_mock, secretsmanager_mock = boto3_mocks
+
+    event = {'pause': True}
+    pause_preservica_activity.lambda_handler(event, None)
+    detail = {"slackMessage": ":alert-noflash-slow: Preservica activity has been paused in environment test"}
+    eventbridge_mock.put_events.assert_called_once_with(
+        Entries=[{
+            "Source": "preservica-activity-pause",
+            "DetailType": "DR2DevMessage",
+            "Detail": json.dumps(detail),
+            "EventBusName": "default"
+        }]
+    )
+
+    eventbridge_mock.disable_rule.assert_called_once_with(Name="test-dr2-entity-event-schedule")
+
+    secretsmanager_mock.cancel_rotate_secret.assert_has_calls([
+        call(SecretId='secret-arn-1'),
+        call(SecretId='secret-arn-2'),
+    ], any_order=True)
+
+
+def test_resume_preservica_activity(env, boto3_mocks):
+    eventbridge_mock, secretsmanager_mock = boto3_mocks
+
+    event = {'pause': False}
+    pause_preservica_activity.lambda_handler(event, None)
+
+    eventbridge_mock.put_events.assert_called_once_with(
+        Entries=[{
+            "Source": "preservica-activity-pause",
+            "DetailType": "DR2DevMessage",
+            "Detail": '{"slackMessage": ":white_check_mark: Preservica activity has been resumed in environment test"}',
+            "EventBusName": "default"
+        }]
+    )
+
+    secretsmanager_mock.rotate_secret.assert_has_calls([
+        call(
+            SecretId='secret-arn-1',
+            RotationLambdaARN='lambda-arn-1',
+            RotationRules={'ScheduleExpression': 'expression1'},
+            RotateImmediately=False
+        ),
+        call(
+            SecretId='secret-arn-2',
+            RotationLambdaARN='lambda-arn-2',
+            RotationRules={'ScheduleExpression': 'expression2'},
+            RotateImmediately=False
+        ),
+    ], any_order=True)
+
+    eventbridge_mock.enable_rule.assert_called_once_with(Name="test-dr2-entity-event-schedule")
+
+
+def test_all_secrets_enabled_rotation_enabled(env, boto3_mocks):
+    eventbridge_mock, secretsmanager_mock = boto3_mocks
+    secretsmanager_mock.describe_secret.side_effect = [
+        {'ARN': 'secret-arn-1', 'RotationEnabled': True},
+        {'ARN': 'secret-arn-2', 'RotationEnabled': True},
+    ]
+    eventbridge_mock.describe_rule.side_effect = [
+        {'State': 'ENABLED'}
+    ]
+    event = {'source': 'aws.events'}
+    pause_preservica_activity.lambda_handler(event, None)
+    eventbridge_mock.put_events.assert_not_called()
+
+
+def test_one_secret_rotation_disabled(env, boto3_mocks):
```

Comment:
Is there a more descriptive name i.e. adding what we expect the this test to do?